### PR TITLE
parser, ddl: support vector index at the parser

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/55694

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: pkg/parser/keywords.go [hunk 1]
```diff
@@ -247,6 +247,7 @@ var Keywords = []KeywordsType{
 	{"VARCHAR", true, "reserved"},
 	{"VARCHARACTER", true, "reserved"},
 	{"VARYING", true, "reserved"},
+	{"VECTOR", true, "reserved"},
```
Comment (tiancaiamao) [inline pkg/parser/keywords.go:250]: Differ from MySQL, this is changed to revered keyword?

**Review by tiancaiamao — APPROVED** [review]

File: pkg/parser/keywords.go (hunk 1 above)
Comment (zimulala) [reply pkg/parser/keywords.go:250]: MySQL does not currently support vector keys: [link](https://dev.mysql.com/doc/refman/9.0/en/vector.html).
Indexing-related keywords such as `fulltext` and `unique` are reserved, so this is also done here. In addition, it is a little complicated to process without a reserved word.

Comment (zimulala) [reply pkg/parser/keywords.go:250]: PM(@joey-yez ) comments that if it is used as the reserved keyword, there will be compatibility. So I will update it.

**Review by joey-yez — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Comment by zimulala** [comment]
PTAL @yudongusa @easonn7

**Review by yudongusa — APPROVED** [review]
LGTM for DDL syntax. Please make sure test this thoroughly.

**Pull request merged**